antlr-kotlin-target/src/main/resources/org/antlr/v4/tool/templates/codegen/Kotlin/Kotlin.stg

```diff
@@ -263,6 +264,7 @@ class <parser.name>(input: TokenStream) : <superClass; null="Parser">(input) {
         <parser.rules:{r | RULE_<r.name>(<r.index>)}; separator=",\n", wrap, anchor>
     }
 
+    @ThreadLocal
```

Comment:
We are generation multi-platform code and therefore I am afraid that this annotation could cause an issue on JVM and JS, right?

Reply:
well perhaps, and honestly it appears it'd mask a problem on linux native to boot. Seeing this @ThreadLocal pattern repeated elsewhere in the samples I perhaps too eagerly submitted this PR too soon. Lets close the PR and I'll work on a more proper solution with thread safe behaviors.